nussl/datasets/hooks.py

```diff
@@ -417,6 +427,146 @@ def process_item(self, item):
+
+        if self.verbose:
+            logging.info(f'Preprocessing data at {self.folder}.')
+
+        items = self.mix_src.get_items(self.folder)
+        balanced_samples = []
+        metadata = []
+        for item in tqdm.tqdm(items, disable=not self.verbose):
+            mix, sources = self.mix_src.get_mix_and_sources(item)
+            target_src = sources[self.salient_src]
+            target_audio = target_src.audio_data
+            salient_starts = utils.find_salient_starts(target_audio,
+                                                       self.segment_dur,
+                                                       self.hop_ratio,
+                                                       mix.sample_rate,
+                                                       self.threshold_db)
+            # this is a fairly cheap operation, no need to embed it within a
+            # conditional
+            balanced_samples.append((len(salient_starts), item,
+                                     salient_starts, mix.sample_rate))
+            for start in salient_starts:
+                if (start + self.segment_dur * mix.sample_rate) <= len(mix):
+                    metadata.append({
+                        'mixsrc_item': item,
+                        'start': start / mix.sample_rate
+                    })
+        # if the balance flag is set then balance the metadata
+        metadata = self._balance_set(balanced_samples) if self.balance \
+            else metadata
+        return metadata
+
+    @staticmethod
+    def _balance_set(sample_counts):
+        """
+        Balance the dataset to contain at least the average number of audio
+        samples
+        Args:
+            sample_counts: counts of the salient samples per audio files
+        Returns:
+        """
+        # TODO: Verify the method of balancing, option 1: oversample until
+        #  everything matches the max or option 2: oversample some under sample
+        #  others
+        avg_length = np.mean([elem[0] for elem in sample_counts])
+        metadata = []
+        for count, song, starts, sample_rate in sample_counts:
+            circular_iterator = 0
+            while circular_iterator <= avg_length:
+                metadata.append({
+                    'mixsrc_item': song,
+                    'start': starts[circular_iterator % count] / sample_rate
+                })
+                circular_iterator += 1
+        return metadata
+
+    def _get_mix(self, placeholder, item):
+        """OnTheFly closure. Gets called for every iteration to build a batch.
+        A placeholder param is necessary since OnTheFly will pass itself as
+        well as the item to this function"""
+        item = self.song_metadata[item]
+        mixsrc_item = item['mixsrc_item']
+        # Note that the onset needs to be passed to the AudioSignal
+        # class as the kwarg offset.
+        onset = item['start']
```

Comment:
Why not rename this variable to `start = item['start']`? That way the code is clearer and we don't need this comment to explain what's happening.

Reply:
As it is right now, this variable is referred to by three names in just 3 lines of code: `'start'` (in `item['start']`), `onset`, and `offset` (in the process_item kwarg). I'm not sure if there's enough context to name the variable `offset` here because there are a few layers of abstraction before the `offset` arg is used directly, so I think `start` conveys what's happening here well.

Sorry for the long winded explanation; I'm walking you through my thought process.